### DISCUSSION TITLE
python tool to get dcp messages

### DISCUSSION THREAD

**Discussion by ghost** [opendcs/opendcs#660] [General]
We just created a Python tool to retrieve GOES DCP messages - https://github.com/dcspy/dcspy. I would love for you to test it and send us your comments.

**Comment by MikeNeilson** [comment]
I'll try to give it a look when I have a chance... though I'd recommend not following the overall class structure we have, that's from 20 years ago before better designs were obvious. I think python already has some built in layers on top of plain sockets to be more "pythonic" and thus be simpler for you to maintain.

That's meant as constructive criticism, not to knock you down. That you created this at all is impressive.

**Reply by ghost** [reply]
Thank you! We did start with `opendcs`, so some design decisions may be left over. But we've refactored a lot to try and make things more "pythonic". There will likely be more refactoring in the future as well.

**Comment by MikeNeilson** [comment]
Also keep an eye on: https://github.com/opendcs/dcs_standards/pull/3

It's definitely not going to replace DDS immediate so you haven't wasted any effort, but we're trying to update systems to modern protocols that are easier to use in say, cloud environments that all the gov users are getting forced into.

I've only done some initial test cases providing the same data over HTTP so now is a good time to get in on helping shape it, if you're interested.
